### PR TITLE
[Payments NOX] e2e drag and drop functionality for providers and offline PMs

### LINKED ISSUE #53247: [Payments NOX] Updates to fully support drag & drop
## Description

In https://github.com/woocommerce/woocommerce/pull/52888, drag & drop components were created.

This issue is to cover the remaining needed changes including:

- Implementing the drag & drop on the main payment methods page.
- Hooking up the drag and drop to fire a POST request to the new endpoint for saving drag&drop order.

### Additional context

In https://github.com/woocommerce/woocommerce/pull/52030, I suggested extracting the drag-and-drop part to a separate issue. 

Discussion about what not to sort: p6riRB-bfm-p2#comment-11466

### Designs
Design post: pfHfG4-q7-p2
Figma: xO8YdpKfaVoP2d8NvweZ5I-fi-1428_21437

### Dev notes
Reactifying the Payments settings page: pe4Hk0-1ly-p2
Populate react-powered main payments settings screen PR: https://github.com/woocommerce/woocommerce/pull/50825

### Additional context
PRD: 1wMfX8CAjDnOp24tcU-PADB0pOt3cz7mGdyZYeRBxx2Y-gdoc

### PR DESCRIPTION
**Pull request by dmallory42**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #53247 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm i && pnpm build`
2. Go to the payment settings page
3. Verify that you can see the drag handles at the left of each payment provider in the list.
4. Open the network tab in the browser console, perform a drag and drop - verify you see a `/order` request in the network tab - verify the request payload looks sensible - the `order_map` should be posted, and it should contain a list of payment providers - the ordering indexes should be swapped, so for example if PayPal has ordering ID 20, and WooPayments has ordering ID 10, and you swap them, the order map request should look like:

```json
{
    "woopayments": 20,
    "paypal": 10,
}
```

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-extension-suggestion-list-item/payment-extension-suggestion-list-item.tsx
Comment (vladolaru) [inline plugins/woocommerce/client/admin/client/settings-payments/components/payment-extension-suggestion-list-item/payment-extension-suggestion-list-item.tsx]:
```suggestion
							label={ __( 'Payment Provider Options', 'woocommerce' ) }
```

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx
Comment (vladolaru) [inline plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx:67]: We need to use the dynamic badge part of the incentive data.

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx
Comment (vladolaru) [inline plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx]:
```suggestion
									'Payment Provider Options',
```

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx
Comment (vladolaru) [inline plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx]: Could we use some centralized constants for these so we would have a single, clear place to update in case the BE changes something?

File: pnpm-lock.yaml
Comment (vladolaru) [inline pnpm-lock.yaml]: Are these package version changes intended?

**Review by vladolaru — APPROVED** [review]
Thank you for working on this, @dmallory42! The testing instructions checked out and my smoke testing likewise. I left some comments, but I am pre-approving it.

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx
Comment (oaratovskyi) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx]: I'll fix it in my PR! https://github.com/woocommerce/woocommerce/pull/53590

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-extension-suggestion-list-item/payment-extension-suggestion-list-item.tsx
Comment (oaratovskyi) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-extension-suggestion-list-item/payment-extension-suggestion-list-item.tsx]: I'll fix it in my PR https://github.com/woocommerce/woocommerce/pull/53590

Comment (oaratovskyi) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-extension-suggestion-list-item/payment-extension-suggestion-list-item.tsx]: done https://github.com/woocommerce/woocommerce/pull/53590/commits/89be2c71fa9da684a938d1f0a53dce141e6721c9

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx
Comment (oaratovskyi) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx]: done https://github.com/woocommerce/woocommerce/pull/53590/commits/89be2c71fa9da684a938d1f0a53dce141e6721c9

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx:67]: I'll address this in https://github.com/woocommerce/woocommerce/pull/53432. 🙂

File: pnpm-lock.yaml
Comment (oaratovskyi) [reply pnpm-lock.yaml]: yeah, looks risky, worth double checking

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx
Comment (vladolaru) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list-item/payment-gateway-list-item.tsx:67]: Oh! Yeah. Mixed up PRs :)

File: plugins/woocommerce/client/admin/client/settings-payments/components/list-placeholder/list-placeholder.scss
Comment (oaratovskyi) [inline plugins/woocommerce/client/admin/client/settings-payments/components/list-placeholder/list-placeholder.scss:1]: seems like something was transferred not fully - see comment in the issue

**Review by oaratovskyi — COMMENTED** [review]
@dmallory42 looking great! 

1. I've noticed that styling of list placeholder changed and now I don't see the top and bottom border. Additionally it feels like it became smaller - oh, because it's jumping and the placeholder items have different padding that actual loaded, see 
https://github.com/user-attachments/assets/790ebfe3-703b-4bb0-bcb0-8fce9ce5d5bf



| Before | After |
|--------|-------|
| Screenshot 2024-12-10 at 16 12 19 | Screenshot 2024-12-10 at 16 11 03 |

2. Also, offline payment list item is clickable on trunk and it has indicator on hover, it's removed in this PR

3. This issue needs to be addressed before merging (it's known one)
Screenshot 2024-12-10 at 16 12 11

happy to approve once you address those!

File: pnpm-lock.yaml
Comment (dmallory42) [reply pnpm-lock.yaml]: No. I have no idea how they keep slipping into my PRs. Has anyone else noticed the same? 🤔

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx
Comment (oaratovskyi) [inline plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx]: `woocommerce-list__item` is typed twice here

I believe we should get `clickable-list-item` back here (see PR review comment)

and full row should be clickable not only icon
```
onClick: () => {
  window.location.href = getAdminLink(
	  'admin.php?page=wc-settings&tab=checkout&section=offline'
  );
  },
```

as it was in `plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.tsx` in this PR in line 101 (sorry, can't get direct link into that)

File: pnpm-lock.yaml
Comment (oaratovskyi) [reply pnpm-lock.yaml]: yeah, probably that is happening when running `pnpm i` - it works like `composer update`
I use now ` pnpm install --frozen-lockfile` - it won't update only install from lockfile!

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx]: Updated in https://github.com/woocommerce/woocommerce/pull/53301/commits/7285d2cdc6aa32ac5607e602ac0ef6033c699594.

File: pnpm-lock.yaml
Comment (dmallory42) [reply pnpm-lock.yaml]: Great, I'll try that. Thank you!

Comment (dmallory42) [reply pnpm-lock.yaml]: Resolved!

File: plugins/woocommerce/client/admin/client/settings-payments/components/list-placeholder/list-placeholder.scss
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/list-placeholder/list-placeholder.scss:1]: Fixed the padding now. Thanks!

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx]: Thank you for spotting this, great catch! I addressed it and it should work as described now.

Comment (vladolaru) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx]: Big nit pick, but do you think we could vertically center the offline PMs group row arrow (like in the designs)?

**Review by oaratovskyi — APPROVED** [review]
Thanks @dmallory42 - it's AMAZING 🚀 🔥 👍 
Works like a charm!

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx
Comment (mordeth) [inline plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx:20]: Nitpick: The type interface could be relocated to `payment-settings/types.ts`.

**Review by mordeth — APPROVED** [review]
The code changes look great, and manual testing was successful!

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx
Comment (oaratovskyi) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx]: addressed in https://github.com/woocommerce/woocommerce/pull/53301/commits/7ec48d27d891852eeb75d6cb9017f125106d1dfb!

Comment (oaratovskyi) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx]: image

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-list/payment-gateway-list.tsx:20]: Thanks for the suggestion! Because the props are only used in this one place, I preferred keeping them in the same file - but in the future we can probably extract a lot of the shared gateway logic and then use a shared types file as you suggest. I will keep it in mind :)

**Pull request merged**
